### DISCUSSION TITLE
Retain or fix transparency from clipboard image

### DISCUSSION THREAD

**Discussion by skygate2012** [ImageMagick/ImageMagick#6091] [Help]
### ImageMagick version

7.1.0

### Operating system, version and so on

Windows 10

### Description

For example, copy any image with transparency in the browser, then execute:
`magick clipboard: output.png`

![bulb](https://user-images.githubusercontent.com/7148976/220246525-eb91506a-ecbf-445b-a3ad-c2e733451265.png)

The result is an image with a black background:

![output](https://user-images.githubusercontent.com/7148976/220246745-1068ef8c-0cce-45ed-a3e0-59991248c780.png)

I tried extracting the alpha channel, and the output is blank.
`magick clipboard: -alpha extract alpha.png`

I read that it's related with Windows using DIB format so third-party programs aren't recognizing the 4th channel as alpha. Is there any workaround for imagemagick to convert this?

**Comment by snibgo** [comment]
As far as I know, Windows loses the alpha data as soon as the image is written to the screen. Windows will do any necessary compositing with what is already on the screen, so what gets written is red, green and blue values only. So capturing a screenshot will get only RGB values.

**Reply by skygate2012** [reply]
It's not lost, just that the Windows DIB format does not follow the standards. [This tool](https://github.com/SilverEzhik/ClipboardToPNG) for example converts DIB to ARGB bitmap.

**Comment by snibgo** [comment]
Sorry, I thought you were asking about screenshots.

If the clipboard contains alpha, IM can read it ("paste" the clipboard image). The difficulty is getting alpha into the clipboard. (Searching gives technical details about DIB and PNG stream.)

IM can write alpha to the clipboard, and read alpha from the clipboard. For example, where `toes_holed.png` has transparency:
```
magick toes_holed.png clipboard:

magick clipboard: x.png
```
... now x.png has transparency.
